### DISCUSSION TITLE
Issues Importing VB6 Project

### DISCUSSION THREAD

**Discussion by dmrvb** [twinbasic/twinbasic#759] [General]
I have just had a first go at importing a very big VB6 project (previous testing has all been copy and paste).
It mostly worked very well and once again I am impressed.
Most issues were GUI related (control references in code modules)
tB again picked up a lot of little errors that VB misses, mostly "default to variant" where I forgot to give a type. Good stuff.

Biggest issue is that "Input" is missing from the File System.   "string = input (n, #stream)" gives "unrecognised symbol" error

vbCancel gives an ambiguous error (DragConstants or vbmsgboxresult)
I can easily fix this but it is a breaking of 100% VB6 compatibility.

Can we have a Screen Object soon please?.
Screen.Mousepointer gave loads of errors.
I am aware of the multi screen discussions here but a basic VB6 compatible Screen. would be good for now, and improve it later?

**Comment by CJCK** [comment]
Similar to dmrvb, we have a decent size commercial application in VB6. I assume several of the people lurking here are in a similar position. To help guide Wayne, I thought it may be helpful if we document what is really preventing us from switching. Then we update our lists as Wayne issues new versions.

Importing our app showed the following:
- Forms not imported.
- Form, Printer and Screen Objects doesn't seem supported.
- Similar to post below, our app also require ctl, pag and res files to be imported (4/4/22).
- Multimonitors at different scales not quite right.

Thanks again for the fantastic effort here Wayne. 

CJCK

**Comment by VB6-Programming** [comment]
Similar to the above posts.

- Forms not importing (.frm) (unsupported document). Also UserControls(.ctl), PropertyPages(.pag), ResFile32(.res) - modules and classes import well.
- Screen.mousepointer (and other Screen object properties)
- vbCancel is ambiguous (DragConstants or vbmsgboxresult)
